soc/arm/st_stm32/stm32wba/power.c

```diff
@@ -8,19 +8,48 @@
 #include <soc.h>
 #include <zephyr/init.h>
 
-#include <stm32wbaxx_ll_utils.h>
 #include <stm32wbaxx_ll_bus.h>
 #include <stm32wbaxx_ll_cortex.h>
 #include <stm32wbaxx_ll_pwr.h>
+#include <stm32wbaxx_ll_icache.h>
 #include <stm32wbaxx_ll_rcc.h>
 #include <stm32wbaxx_ll_system.h>
 #include <clock_control/clock_stm32_ll_common.h>
 
+#ifdef CONFIG_BT_STM32WBA
+#include "scm.h"
+#endif
+
 #include <zephyr/logging/log.h>
+
 LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
 
 void set_mode_stop(uint8_t substate_id)
 {
+
+	LL_PWR_ClearFlag_STOP();
+	LL_RCC_ClearResetFlags();
+
+	/* Disabling ICACHE is required to properly enter stop mode */
```

Comment:
what about testing  LL_ICACHE_IsEnabled() to  decide to disable and to wait on ICACHE flag

Reply:
Indeed, fixed.